core/controllers/diagnostic_test_player_test.py

```diff
@@ -36,40 +35,41 @@ def setUp(self) -> None:
         super().setUp()
         self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
         self.owner_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+        self.original_parameter_registry = (
+            platform_parameter_registry.Registry.parameter_registry.copy())
 
+    def tearDown(self) -> None:
+        super().tearDown()
+        platform_parameter_registry.Registry.parameter_registry = (
+            self.original_parameter_registry)
+
+    def test_should_not_access_diagnostic_test_page_when_feature_is_disabled(
+        self) -> None:
         feature_services.update_feature_flag(
             platform_parameter_list.ParamNames.DIAGNOSTIC_TEST.value,
             self.owner_id,
             'test update',
-            [
-                platform_parameter_domain.PlatformParameterRule.from_dict({
-                    'filters': [{
-                        'type': 'server_mode',
-                        'conditions': [[
-                            '=',
-                            platform_parameter_domain.FeatureStages.DEV.value]
-                            ]
-                    }],
-                    'value_when_matched': True
-                })
-            ],
+            [],
             False
         )
+        self.get_html_response(
+            feconf.DIAGNOSTIC_TEST_PLAYER_PAGE_URL,
+            expected_status_int=404
+        )
 
-    def test_should_access_diagnostic_test_page_in_dev_mode(self) -> None:
-        with self.swap(constants, 'DEV_MODE', True):
-            self.get_html_response(
-                feconf.DIAGNOSTIC_TEST_PLAYER_PAGE_URL,
-                expected_status_int=200
-            )
-
-    def test_should_not_access_diagnostic_test_page_when_not_in_dev_mode(
+    def test_should_access_diagnostic_test_page_when_feature_is_enabled(
         self) -> None:
-        with self.swap(constants, 'DEV_MODE', False):
-            self.get_html_response(
-                feconf.DIAGNOSTIC_TEST_PLAYER_PAGE_URL,
-                expected_status_int=404
-            )
+        feature_services.update_feature_flag(
```

Comment:
This test (and quite a few of the other tests you've edited) should have a pre-check.

Reply:
I have added the precheck to the tests I have added.